src/internal.c

```diff
@@ -30021,8 +30047,17 @@ static int DoSessionTicket(WOLFSSL* ssl, const byte* input, word32* inOutIdx,
         if (ret == 0 && ssl->options.dtls)
             DtlsMsgPoolReset(ssl);
 #endif
-        WOLFSSL_LEAVE("DoClientHello", ret);
-        WOLFSSL_END(WC_FUNC_CLIENT_HELLO_DO);
+
+        if (ret == 0) {
+            /* Set correct options for this connection */
```

Comment:
These were not being set?

Reply:
These were being set incorrectly in the cipher selection (although that may be due to the context setting error above)? I will need to confirm this.

Reply:
This was not necessary. Fixing the `ssl->version` modification fixed these options as well.